src/pymorize/config.py

```diff
@@ -0,0 +1,225 @@
+"""
+This module defines the configuration hierarchy for the pymorize application, using
+``everett``'s ``~everett.manager.ConfigManager``. The configuration hierarchy is as follows (lowest to highest
+priority):
+    1. Hardcoded defaults
+    2. User configuration file
+    3. Run-specific configuration
+    4. Environment variables
+    5. Command-line switches
+
+The configuration hierarchy is defined in the ``from_pymorize_cfg`` class method, and
+cannot be modified outside the class. You should initialize a ``PymorizeConfigManager``
+object (probably in your ``CMORizer``) and grab config values from it by calling with the
+config key as an argument.
+
+User Configuration File
+-----------------------
+
+You can define global configuration options in a user configuration file. The files found at these
+locations will be used, in highest to lowest priority order:
+    1. ``${PYMORIZE_CONFIG_FILE}``
+    2. ``${XDG_CONFIG_HOME}/pymorize.yaml``
+    3. ``${XDG_CONFIG_HOME}/pymorize/pymorize.yaml``
+    4. ``~/.pymorize.yaml``
+
+Note that the ``${XDG_CONFIG_HOME}`` environment variable defaults to ``~/.config`` if it is not set.
+
+Configuration Options
+---------------------
+
+You can configuration the following:
+
+.. autocomponentconfig:: pymorize.config.PymorizeConfig
+   :case: upper
+   :show-table:
+   :namespace: pymorize
+
+Usage
+-----
+Here are some examples of how to use the configuration manager::
+
+    >>> pymorize_cfg = {}
+    >>> config = PymorizeConfigManager.from_pymorize_cfg(pymorize_cfg)
+
+    >>> engine = config("xarray_backend")
+    >>> print(f"Using xarray backend: {engine}")
+    'netcdf4'
+
+    >>> parallel = config("parallel")
+    >>> print(f"Running in parallel: {parallel}")
+    True
+
+You can define a user file at ``${XDG_CONFIG_DIR}/pymorize/pymorize.yaml``::
+
+    >>> import pathlib
+    >>> cfg_file = pathlib.Path("~/.config/pymorize/pymorize.yaml").expanduser()
+    >>> cfg_file.mkdir(parents=True, exist_ok=True)
+    >>> with open(cfg_file, "w") as f:
+    ...     f.write("xarray_backend: zarr\nparallel: False\n")
+    >>> config = PymorizeConfigManager.from_pymorize_cfg()
+    >>> engine = config("xarray_backend")
+    >>> print(f"Using xarray backend: {engine}")
+    'zarr'
+    
+
+See Also
+--------
+- `Everett Documentation <https://everett.readthedocs.io/en/latest/>`_
+"""
+
+import os
+import pathlib
+
+from everett import InvalidKeyError
+from everett.ext.yamlfile import ConfigYamlEnv
+from everett.manager import (ChoiceOf, ConfigDictEnv, ConfigManager,
+                             ConfigOSEnv, Option, _get_component_name,
+                             parse_bool)
+
+
+def _parse_bool(value):
+    if isinstance(value, bool):
+        return value
+    return parse_bool(value)
+
+
+class PymorizeConfig:
+    class Config:
+        quiet = Option(
+            default="no", doc="Whether to suppress output.", parser=_parse_bool
```

Comment:
I think I am starting to understand... but I really don't like to train the users into thinking that if they define `parameter: "no"` in a yaml file they are saying always `False` for every tool out there... also yaml "understands" true/false, so... it is giving me the creeps... sorry... I could learn to live with `yes`/`no` if necessary hehe